instrumentation/opentelemetry-instrumentation-openai-v2/src/opentelemetry/instrumentation/openai_v2/utils.py

```diff
@@ -43,69 +48,138 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
-def extract_content(choice):
-    if getattr(choice, "message", None) is None:
-        return ""
-
-    # Check if choice.message exists and has a content attribute
-    message = choice.message
-    if getattr(message, "content", None):
-        return choice.message.content
-
-    # Check if choice.message has tool_calls and extract information accordingly
-    elif getattr(message, "tool_calls", None):
-        result = [
-            {
-                "id": tool_call.id,
-                "type": tool_call.type,
-                "function": {
-                    "name": tool_call.function.name,
-                    "arguments": tool_call.function.arguments,
-                },
-            }
-            for tool_call in choice.message.tool_calls
-        ]
-        return result
-
-    # Check if choice.message has a function_call and extract information accordingly
-    elif getattr(message, "function_call", None):
-        return {
-            "name": choice.message.function_call.name,
-            "arguments": choice.message.function_call.arguments,
-        }
+def extract_content_from_prompt_message(message) -> Tuple[str, Any]:
+    content = get_property_value(message, "content")
+    if content:
+        return ("content", content)
+
+    tool_calls = extract_tool_calls(message)
+    if tool_calls:
+        return ("tool_calls", tool_calls)
 
-    # Return an empty string if none of the above conditions are met
     else:
-        return ""
+        return None
 
 
-def extract_tools_prompt(item):
-    tool_calls = getattr(item, "tool_calls", None)
+def extract_tool_calls(item):
+    tool_calls = get_property_value(item, "tool_calls")
     if tool_calls is None:
         return
 
     calls = []
     for tool_call in tool_calls:
         tool_call_dict = {
-            "id": getattr(tool_call, "id", ""),
-            "type": getattr(tool_call, "type", ""),
+            "id": get_property_value(tool_call, "id"),
+            "type": get_property_value(tool_call, "type"),
         }
 
-        if hasattr(tool_call, "function"):
-            tool_call_dict["function"] = {
-                "name": getattr(tool_call.function, "name", ""),
-                "arguments": getattr(tool_call.function, "arguments", ""),
-            }
+        func = get_property_value(tool_call, "function")
+        if func:
+            tool_call_dict["function"] = {}
+
+            name = get_property_value(func, "name")
+            if name:
+                tool_call_dict["function"]["name"] = name
+
+            arguments = get_property_value(func, "arguments")
+            if arguments:
+                tool_call_dict["function"]["arguments"] = arguments.replace(
```

Comment:
From the definition of [Function object](https://github.com/open-telemetry/semantic-conventions/blob/v1.28.0/docs/gen-ai/gen-ai-events.md#function-object), it seems like `arguments` can be `AnyValue` so `replace` would probably error out in those cases. Perhaps we simply leave the value as is?

Reply:
`arguments` is a string in openai REST api and client.  Semconv leave it undefined for the future. I can check if it's a sting - I don't want to have new lines in the logs, so replacing is kinda important